### PR TITLE
[domain] Don't add NULL runtime to runtimes list

### PR DESCRIPTION
**Pull request by lambdageek**
Also when iterating over the runtimes, actually look at the current element,
not the first element.

Also mention the unsupported runtime version in the warning message when it is
passed to mono_init_internal.

Fixes https://github.com/mono/mono/issues/17916

### CODE REVIEW

File: mono/metadata/domain.c [hunk 1]
```diff
@@ -590,20 +590,27 @@ mono_init_internal (const char *filename, const char *exe_filename, const char *
 		mono_fixup_exe_image (exe_image);
 #endif
 	} else if (runtime_version != NULL) {
-		runtimes = g_slist_prepend (runtimes, (gpointer)get_runtime_by_version (runtime_version));
+		const MonoRuntimeInfo* rt = get_runtime_by_version (runtime_version);
+		if (rt != NULL)
```
Comment (CoffeeFlux) [inline mono/metadata/domain.c]: Should we not warn here as well?

Comment (lambdageek) [reply mono/metadata/domain.c]: If `rt` is null here, `runtimes` will be null, so we will fall down into the default runtime code which will warn.

That is, `mono_init_version ("pedump", "v2.0.50727")` will print out `WARNING: The requested runtime version "v2.0.50727" is unavailable`.

Comment (CoffeeFlux) [reply mono/metadata/domain.c]: Ah okay, misread surrounding code.

**Comment by lambdageek** [comment]
@monojenkins backport to 2019-12

**Comment by lambdageek** [comment]
@monojenkins backport to 2019-10

**Pull request merged**